sdk/python/lib/pulumi/runtime/invoke.py

```diff
@@ -285,8 +297,12 @@ def do_invoke():
                 deserialized, lambda prop: prop, typ
             )
 
+        dependencies: List["Resource"] = []
+        for _, property_deps in property_dependencies.items():
+            for dep in property_deps:
+                dependencies.append(dep)
         return (
-            InvokeResult(result, is_secret),
+            InvokeResult(value=result, is_secret=is_secret, dependencies=dependencies),
```

Comment:
Don't we also have to track if any of the serialized inputs were a secret, the output should also be a secret?

If we add a `keep_secrets` option to `serialize_properties` and are setting it to False, then there'd be no way after calling it to know if there were any serialized secrets, unless `serialize_properties` was augmented in some way to report whether it saw any secrets.

Or we leave `serialize_properties` as-is and let it serialize secrets to the special secret object, and then do a pass after calling it that looks for those and replaces them with just the value, and then knows that there was a secret. Simpler as it doesn't involve modifying `serialize_properties` but it does mean we have to do another pass over the object graph to fix it up, which could theoretically possibly could have a perf impact for large objects.

Reply:
> unless serialize_properties was augmented in some way to report whether it saw any secrets.

We could have something like a `on_secret_encountered` callback but but that would be added in combination with `keep_secrets`. I much rather prefer the second alternative of doing another pass on the object graph. I don't think the perf impact here is negligible, invoke inputs tend to be really small objects

Reply:
Opted for the latter option, implemented as follows:
```py
inputs_contain_secrets = False
def on_secrets_encountered():
    nonlocal inputs_contain_secrets
    inputs_contain_secrets = True
invoke_args = rpc.unwrap_rpc_secret_properties(inputs, on_secrets_encountered)
```